drivers/serial.c

```diff
@@ -137,3 +138,24 @@ int serial_write(io_port_t port, const char *buf, size_t len) {
 
     return rc;
 }
+
+void uart_input_enable(void) {
+    int unmasked_irqs = inb(PIC1_PORT_DATA);
+    int unmask_com = COM1_IRQ_UNMASK & COM2_IRQ_UNMASK & unmasked_irqs;
+
+    outb(PIC1_PORT_DATA, unmask_com); /* unmask com irqs */
+}
+
+void uart_handler(io_port_t ports[2]) {
+    io_port_t port;
+    unsigned int i, status, data_avail;
+    outb(PIC1_PORT_CMD, PIC_EOI);
+    for (i = 0; i < sizeof(*ports); ++i) {
+        port = ports[i];
+        status = inb(port + UART_IIR_REG_OFFSET);
+        data_avail = (status & UART_IIR_STATUS_MASK) == UART_IIR_RBR_READY;
+        if (data_avail) {
+            printk("%c", inb(port + UART_RBR_REG_OFFSET));
```

Comment:
It might make sense to use a small cyclic buffer for storing passed data (let's say size 80 chars). It can be helpful for later functionality built on top of this.